### DISCUSSION TITLE
mqtt publish message for Keypad XHK1-UE is wrong

### DISCUSSION THREAD

**Discussion by barbaschela** [Koenkk/zigbee2mqtt#10671] [Devices]
Hi all,
Alarm keypads they have stopped working with Zigbee2mqtt  version 1.22.2-1 addon HA.
The zigbee message is correct but the MQTT does not extract the values ​​correctly  for "action" or "action_code"

Actually log (malfunction):
Info MQTT publish: topic 'zigbee2mqtt/Teclado2', payload '{"action":null,"action_code":null,"action_transaction":null,"action_zone":null,"battery":100,"battery_low":false,"contact":true,"last_seen":"2022-01-09T12:14:31+01:00","linkquality":123,"occupancy":true,"presence":null,"tamper":false,"temperature":19,"voltage":6000}'

Old time log (correct) the action code is a example ficticious:
Info MQTT publish: topic 'zigbee2mqtt/Teclado2', payload '{"action":arm_all_zones,"action_code":2598,"action_transaction":null,"action_zone":null,"battery":100,"battery_low":false,"contact":true,"last_seen":"2021-12-24T10:15:32+01:00","linkquality":123,"occupancy":true,"presence":null,"tamper":false,"temperature":18,"voltage":6000}'

Extract log
Debug Received Zigbee message from 'Teclado2', type 'commandGetPanelStatus', cluster 'ssIasAce', data '{}' from endpoint 1 with groupID 0
Info MQTT publish: topic 'zigbee2mqtt/Teclado2', payload '{"action":null,"action_code":null,"action_transaction":null,"action_zone":null,"battery":100,"battery_low":false,"contact":true,"last_seen":"2022-01-09T12:14:28+01:00","linkquality":120,"occupancy":true,"presence":null,"tamper":false,"temperature":19,"voltage":6000}'
_**Debug Received Zigbee message from 'Teclado2', type 'commandArm', cluster 'ssIasAce', data '{"armmode":3,"code":"2287","zoneid":23}' from endpoint 1 with groupID 0**_
Info MQTT publish: topic 'zigbee2mqtt/Teclado2', payload '{"action":null,"action_code":null,"action_transaction":null,"action_zone":null,"battery":100,"battery_low":false,"contact":true,"last_seen":"2022-01-09T12:14:31+01:00","linkquality":123,"occupancy":true,"presence":null,"tamper":false,"temperature":19,"voltage":6000}'
Info MQTT publish: topic 'zigbee2mqtt/Teclado2', payload '{"action":null,"action_code":null,"action_transaction":null,"action_zone":null,"battery":100,"battery_low":false,"contact":true,"last_seen":"2022-01-09T12:14:31+01:00","linkquality":123,"occupancy":false,"presence":null,"tamper":false,"temperature":19,"voltage":6000}'

Sorry for my english, thanks in advanced

**Comment by Koenkk** [comment]
See ##10504

**Reply by barbaschela** [reply]
Thanks, very fast Koenk, sorry for not find it
